### PR TITLE
fix(cli): prevent duplicate snapshots when --all and a source path are given

### LINKED ISSUE #3057: "snapshot create" creates two snapshots?
```
root@nas:~# kopia --version
0.13.0 build: 7f69502bddd6650e7cb7a132291bc54920e58988 from: kopia/kopia
```

I have a cron job that creates a snapshot every night:

`13 1 * * * /usr/bin/kopia snapshot create --all --log-level=info --no-progress /zfs`

And every night, it logs what looks to be like it creating two snapshots:

```
Snapshotting root@nas:/zfs ...
Created snapshot with root kf49bbe04f7bcb00e0474f42f55236d25 and ID dcf55fa5b99f187a1bc8333c7767a22a in 5h39m35s
Snapshotting root@nas:/zfs ...
Created snapshot with root k30b0c795950f0c0e82d6f9ac4de69abd and ID 92b4030916d3a9031a76ebb523de4604 in 2h26m28s
```

This is what it looks like in the GUI:

Capture

Is this expected?

### PR DESCRIPTION
**Pull request by NickIAm**
Throw an error if using --all with a source path as it will create a  duplicate snapshot when the same source is listed twice. Fixes #3057

### CODE REVIEW

**Comment by jkowalski** [comment]
please add a test for this behavior.

**Comment by NickIAm** [comment]
Added a test function in snapshot_create_test.go

File: cli/command_snapshot_create.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func (c *commandSnapshotCreate) setup(svc appServices, parent commandParent) {
 func (c *commandSnapshotCreate) run(ctx context.Context, rep repo.RepositoryWriter) error {
 	sources := c.snapshotCreateSources
 
+	if c.snapshotCreateAll && len(sources) > 0 {
+		return errors.New("cannot use --all with a source path given")
```
Comment (julio-lopez) [inline cli/command_snapshot_create.go:93]:
```suggestion
		return errors.New("cannot use --all when a source path argument is specified")
```

File: cli/command_snapshot_create.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func (c *commandSnapshotCreate) setup(svc appServices, parent commandParent) {
 	cmd := parent.Command("create", "Creates a snapshot of local directory or file.")
 
 	cmd.Arg("source", "Files or directories to create snapshot(s) of.").StringsVar(&c.snapshotCreateSources)
-	cmd.Flag("all", "Create snapshots for files or directories previously backed up by this user on this computer").BoolVar(&c.snapshotCreateAll)
+	cmd.Flag("all", "Create snapshots for files or directories previously backed up by this user on this computer. Cannot be used when a source path is also given.").BoolVar(&c.snapshotCreateAll)
```
Comment (julio-lopez) [inline cli/command_snapshot_create.go:58]:
```suggestion
	cmd.Flag("all", "Create snapshots for files or directories previously backed up by this user on this computer. Cannot be used when a source path argument is also specified.").BoolVar(&c.snapshotCreateAll)
```

**Review by julio-lopez — APPROVED** [review]

**Pull request merged**